### ISSUE TITLE
Getting the error in CircleCI "CypressError: Cypress detected a cross origin error happened on page load" for same domain

### ISSUE THREAD

**Issue by olefirenko** [cypress-io/cypress#2448]
### Current behavior:
The current code `cy.contains('Pay for booking').click()` returns the error in CircleCI:
```
CypressError: Cypress detected a cross origin error happened on page load:

  > Blocked a frame with origin "http://agents.test" from accessing a cross-origin frame.

Before the page load, you were bound to the origin policy:
  > http://agents.test

at Object.cypressErr (http://agents.test/__cypress/runner/cypress_runner.js:68040:11)
      at Object.throwErr (http://agents.test/__cypress/runner/cypress_runner.js:68005:18)
      at Object.throwErrByPath (http://agents.test/__cypress/runner/cypress_runner.js:68032:17)
      at onPageLoadErr (http://agents.test/__cypress/runner/cypress_runner.js:58363:21)
      at HTMLIFrameElement.<anonymous> (http://agents.test/__cypress/runner/cypress_runner.js:63921:19)
      at HTMLIFrameElement.dispatch (http://agents.test/__cypress/runner/cypress_runner.js:21476:27)
      at HTMLIFrameElement.elemData.handle (http://agents.test/__cypress/runner/cypress_runner.js:21288:28)
```

But when I run the tests locally everything is ok. As you can see the domain is the same.
The URL from where I click the button is `http://agents.test/en/booking_details/72MRGKJ` and the destination URL is `http://agents.test/adyen/checkout/9dc3c07d-18fc-4322-ba10-d390ff3234a7`
How is Cypress defining the CORS violation? Could it be the items in Response headers or cookies?

### Desired behavior:

Tests should pass.

### Versions

CircleCI (The server is http://php.net/manual/en/features.commandline.webserver.php)
OS: Linux Ubuntu - 18.04
Browser: Electron 59.0.3071.115
Cypress: v3.1.0

**Comment by jennifer-shehane** [comment]
Do you have a screenshot/video of the failure? Is there anything strange being printed in the command log (the lefthand side of Cypress commands)?

**Comment by olefirenko** [comment]
@jennifer-shehane You can find the screenshot here https://monosnap.com/file/pKBg4hQBNitQel2lbv0V6cv7tluWMU
Also if it helps, when I added `"chromeWebSecurity": false` the url changed to `data:text/html,chromewebdata` https://monosnap.com/file/vB6JWfxTUQYmbUpkJ6ANl3yu6vKrB5

**Comment by itaykotler-fundbox** [comment]
@jennifer-shehane, this happens to me as well 😞
OS: macOS High Sierra, version: 10.13.6 (17G65)
Browser: **Electron 59.0.3071.115**
Cypress: v3.1.0
`"chromeWebSecurity"` is set to `false`.

Works on Chrome Version 69.0.3497.100 (Official Build) (64-bit)

**Comment by grantspeelman** [comment]
@jennifer-shehane same thing happening to me :(
`cypress.json`
```json
{
  "baseUrl": "http://localhost:3002",
  "defaultCommandTimeout": 10000,
  "chromeWebSecurity": false
}
```
Let me know if there is anything I can do to help debug this.

**Comment by grantspeelman** [comment]
I wonder if this related
From https://bugs.chromium.org/p/chromedriver/issues/detail?id=1272
```
Issue Description:
On connection error, GET url returns "data:text/html,chromewebdata".

Steps to reproduce (if relevant, you MUST provide a simplified html page or
link to public site):
1. Navigate to a URL which produces ERR_CONNECTION_REFUSED in Chrome.
2. GET url for that window and see "data:text/html,chromewebdata"
```

**Comment by jennifer-shehane** [comment]
Could you try to update to version [3.1.2](https://on.cypress.io/changelog#3-1-2) as we just fixed some cross origin error issues. 

Otherwise we'll have to close this issue if no reproducible example is provided. Can anyone provide a way to reproduce this?

**Comment by egucciar** [comment]
this is happening to me as well - very inconsistent. I dont believe i can provide a "reproducible example" either. its occuring on 3.1.4 for me in electron as well :(

**Comment by egucciar** [comment]
ooph: 
![image](https://user-images.githubusercontent.com/2848185/51347727-e3146480-1a6e-11e9-9560-5500aba05cdc.png)

**Comment by egucciar** [comment]
My guess is that trying to access properties from other frames is causing an issue. I can see the same code break in certain lines of cypress runner code - e.g. when cypress is getting window props and needs to do `contentWindow.document` it fails there, as well. So it seems, trying to access properties from a frame other than your own is throwing chrome bugs now. Which is weird especially since it is stemming from cypress code. It appears I am not doing anything extremely weird in my code either. For now I've disabled chrome security which serves as a workaround.  I'm going to re-search through the other github issues since I'm definitely confused about this one, and why I'm suddenly running into it.

**Comment by xavivars** [comment]
I'm facing exactly the same error, but no idea how to reproduce it in a way I can share it here.

**Comment by jennifer-shehane** [comment]
Unfortunately we have to close this issue as there is not enough information to reproduce the problem. 

Please comment in this issue with a reproducible example and we will reopen the issue. 🙏

**Comment by Deemon47** [comment]
@jennifer-shehane, Hi, I have same issue, when i try to change browser's date by [this example](https://docs.cypress.io/api/commands/clock.html#Now)



<details><summary> The error is</summary>


```
Cypress detected that an uncaught error was thrown from a cross origin script.

We cannot provide you the stack trace, line number, or file where this error occurred.

Check your Developer Tools Console for the actual error - it should be printed there.

It's possible to enable debugging these scripts by adding the 'crossorigin' attribute and setting a CORS header.

https://on.cypress.io/cross-origin-script-error

This error originated from your application code, not from Cypress.

When Cypress detects uncaught errors originating from your application it will automatically fail the current test.

This behavior is configurable, and you can choose to turn this off by listening to the 'uncaught:exception' event.

https://on.cypress.io/uncaught-exception-from-application
    at createErrFromMsg (http://simpla.v2/__cypress/runner/cypress_runner.js:58811:14)
    at Object.createUncaughtException (http://simpla.v2/__cypress/runner/cypress_runner.js:58818:13)
    at Object.onUncaughtException (http://simpla.v2/__cypress/runner/cypress_runner.js:61698:20)
    at onError (http://simpla.v2/__cypress/runner/cypress_runner.js:61197:39)
From previous event:
    at run (http://simpla.v2/__cypress/runner/cypress_runner.js:61382:15)
    at Object.cy.(anonymous function) [as clock] (http://simpla.v2/__cypress/runner/cypress_runner.js:61610:11)
    at Context.runnable.fn (http://simpla.v2/__cypress/runner/cypress_runner.js:61740:20)
    at callFn (http://simpla.v2/__cypress/runner/cypress_runner.js:30560:21)
    at Test.Runnable.run (http://simpla.v2/__cypress/runner/cypress_runner.js:30553:7)
    at http://simpla.v2/__cypress/runner/cypress_runner.js:64685:28
From previous event:
    at Object.onRunnableRun (http://simpla.v2/__cypress/runner/cypress_runner.js:64680:20)
    at $Cypress.action (http://simpla.v2/__cypress/runner/cypress_runner.js:60037:51)
    at Test.Runnable.run (http://simpla.v2/__cypress/runner/cypress_runner.js:63746:20)
    at Runner.runTest (http://simpla.v2/__cypress/runner/cypress_runner.js:31023:10)
    at http://simpla.v2/__cypress/runner/cypress_runner.js:31129:12
    at next (http://simpla.v2/__cypress/runner/cypress_runner.js:30943:14)
    at http://simpla.v2/__cypress/runner/cypress_runner.js:30953:7
    at next (http://simpla.v2/__cypress/runner/cypress_runner.js:30885:14)
    at http://simpla.v2/__cypress/runner/cypress_runner.js:30921:5
    at timeslice (http://simpla.v2/__cypress/runner/cypress_runner.js:26162:27)
```

</details>


Cypress version: v3.2.0
OS: Linux Mint 19.1 Cinnamon
Browser: Chrome  v73.0.3683.

```js

describe("Date time",()=>{
    it("should not fails",()=>{
        const now = new Date(2017, 3, 14).getTime() // April 14, 2017 timestamp
        cy.clock(now)
        cy.visit('/');
    })
})
```

**Comment by jennifer-shehane** [comment]
@Deemon47 I'm unable to reproduce this when calling `cy.visit('https://example.cypress.io')` in the example you provided. This failure is likely somehow specific to the actual url you are visiting in `cy.visit()`. 

Can you provide the url here? I think an error is being thrown somewhere from the application code (or a 3rd party dep of the app).

**Comment by GeeWee** [comment]
I can confirm this happened to me today multiple times, but after an hour or so it simply disappeared, and I've been unable to reproduce it since.

**Comment by Deemon47** [comment]
@jennifer-shehane , Thanks for response.
I runs tests on my localhost, so i can't provide the url. 

Today strange things happen: After my post i wasn't touch the code, and now when I run the test again, there is no `cross-origin` error. Now it has `Yandex Maps API` error - it's used on my project and looks like this API checks the current date and throw an exception when date is wrong. I understand that current error it is not `cypress` error, but why there was `cross-origin` error at previous time?  look like bug.

**Comment by gerardovanegas-eagle** [comment]
Hello! This is my code, I hope this one is reproducible:

describe('Happy Path B2C / Checkout V2 ', function () {
    beforeEach("Home - Catalog - PP - Cart", function(){

            //Going directly to product's page 
            cy.visit('https://juntozstagingv2.com/es-pe/');

            //Closing Ads
            cy.get("div[data-action='close-mc-modal']").click({force:true})

            //Triggerin Login
            cy.get("a[aria-haspopup='true']").click();
            cy.get("a[href='https://juntozstagingv2.com/account/login']").click();
        })
    });

I'm not sure if this is the issue in the topic but I get this screen when I try to access certain pages:
![image](https://user-images.githubusercontent.com/45433407/57877422-6f485580-77dd-11e9-80eb-67dfdaeab9cd.png)
This I can easily access on the same browser used by cypress
![image](https://user-images.githubusercontent.com/45433407/57877469-8d15ba80-77dd-11e9-956c-4473ab472661.png)

I have my "chromeWebSecurity": false on my cypress file just in case

**Comment by VoloshinS** [comment]
Strange thing, but it seems, I've managed to fix this issue for my case by just cleaning site data in "cypress Chrome".

I've been working on the separate simple static server for CI (which meant to replace webpack-dev-server). So each time when I've been switching to a new server (which served from the same URL "http://localhost:3000" as webpack-dev-server) and rerun tests, I've encountered issue described here (`CypressError: Cypress detected a cross-origin error happened on page load`)
It happened when tests were trying to get fallback for SPA paths. Something like this:
```js
app.use('*', (req, res) => {  res.sendFile('index.html') })
``
I thought the problem was in fallback implementation.
But after cleaning site data issue just disappeared.
I've been trying to reproduce it to find out the reason, but without success.

**Comment by gabbersepp** [comment]
Is someone of you using the cypress retry plugin? I can reproduce this in our codebase but only if I use the retry plugin

**Comment by gabbersepp** [comment]
Not sure where the discussion should be continued. So i copy & paste the same into this issue:

ok here is a small example:
https://github.com/gabbersepp/cypress-example/tree/cy-issue-2448

StR:

bash 1: npm run server
bash 2: npm run cypress:open
execute "cross" spec
please note: it seems that using the cypress retry plugin is the key in our case.
I try to find a bugfix for our usecase.

**Comment by gabbersepp** [comment]
The bug occurrs when executing `cy.wrap().as()` while trying to get the remote jquery instance: 

![image](https://user-images.githubusercontent.com/3234900/69149008-dbca4700-0ad5-11ea-8c7b-73ac9b7bdb0d.png)

![image](https://user-images.githubusercontent.com/3234900/69149025-e553af00-0ad5-11ea-91e4-f1d225659969.png)

**Comment by gabbersepp** [comment]
I tried to avoid the usage of `cy.wrap()` but this did not help. I will open a issue in the retry repository

**Comment by Postavshik** [comment]
> We were seeing seemingly random, intermittent CORS errors with the same domain listed, much like the original issue here. I tried all the various solutions, nothing worked — we'd still see occasional CORS errors. Seemed like an obvious race condition having to do with our DB setup, and finally I tracked it down...
> 
> In our `beforeEach` we seed our DB, and I realized that in a few cases we were ending our tests with a `.click()` on an element that would trigger a navigation. However the test runner would move on to the next test, running `beforeEach` which would teardown the DB state _before the page loaded from the prior click_ which in app caused an error when the page did load, resulting in a redirect! **The solution for us was to ensure that tests didn't end with an action that caused navigation.**
> 
> This is probably an edge case, but figured I'd add it here in case it might be helpful to anyone, since we banged our heads against this for a long time.

Yup... I had exactly the same scenario as yours and it took me a day to figure this out. A fix was to add an assertion after a click command that URL contains certain expected text. That fixed the issue because now cypress was waiting to complete click() command before running beforeEach for the next test.

**Comment by jmwohl** [comment]
@Postavshik Ah, good call!

**Comment by Just01** [comment]
> @jennifer-shehane same thing happening to me :(
> `cypress.json`
> 
> ```json
> {
>   "baseUrl": "http://localhost:3002",
>   "defaultCommandTimeout": 10000,
>   "chromeWebSecurity": false
> }
> ```
> 
> Let me know if there is anything I can do to help debug this.

Thank you! This helps for me) I have successfully implemented  login and logout with cypress + keycloak <3

**Comment by brettz9** [comment]
Because my server threw an exception that was not caught during one of my tests, causing the server to go offline before the test completed (and the test tried to reach the server again), I got this error.

**Comment by OliverJAsh** [comment]
We were also running into this with a login form. We discovered that when the form was submitted, the `POST` request redirected to `http://localhost:5000/__/` (the Cypress test runner) when it should have redirected to `http://localhost:5000/` (our app). When we fixed the redirect `location`, this issue disappeared.

Perhaps Cypress could improve the way this error is displayed. Reading the error and the docs, it doesn't make any sense why we would get a cross origin error when the origins are actually the same. I still don't understand why we would get a cross origin error just because we redirected to the wrong page, because it's still the same origin.

**Comment by rrauenza** [comment]
Also seeing this,

```
CypressError: Cypress detected a cross origin error happened on page load:
  > Blocked a frame with origin "http://localhost:47064" from accessing a cross-origin frame.
Before the page load, you were bound to the origin policy:
  > http://localhost:47064
A cross origin error happens when your application navigates to a new URL which does not match the origin policy above.


```

`Because this error occurred during a `before each` hook we are skipping the remaining tests in the current suite: `Team e2e`
`

Now.. I've been monkeying around with my beforeEach and added some cy.intercepts recently to deal with the problem that my cy.request() with followRedirect=false not always being honored.

```
function do_login(username: string = 'e2e', password: string = 'e2e'): void {
  // ignore any GET's during login due to cy.request not honoring followRedirect always.
  cy.intercept(
    {method: 'GET'},
    {
      statusCode: 400,
    }
  );
  cy.request({method: 'POST', url: 'api/rest/accounts/login/', followRedirect: false, body: {username: username, password: password}}).then(
    (response) => {
      expect(response.status).to.eq(200);
      const jwt = JSON.stringify(response.body.jwt);
      window.localStorage.setItem('jwt', jwt);
      expect(window.localStorage.getItem('jwt')).to.eq(jwt);
    }
  );
  // re-enable GET's
  cy.intercept(
    {method: 'GET'},
    (req) => {
      req.continue((res) => {})
    }
  );
}
```

**Comment by lpyfm** [comment]
I had the same error in GitLab CI and cypress 4.0.1. Our e2e-tests sometimes succeeded and sometimes failed in GitLab, but always succeeded when run on a developer machine. It turns out that the issue was insufficient shared memory: Monitoring the `/dev/shm` file showed that, when the test cases failed, no (or close to none) shared memory was available. I was able to solve the issue by adding the following in `/etc/gitlab-runner/config.toml`:
```yaml
[[runners]]
# …
  [runners.docker]
    # …
    shm_size = 536870912
```
Of course, 536870912 is just an example value, choose whatever you think is sensible. The important part is that you increase it from the default 16 MB.

I was very surprised that insufficient `shm_size` causes cross-origin errors, but I've repeated the tests with and without this fix a few times, so I'm very confident that this was the root issue in our case.

**Comment by papb** [comment]
Hello everyone, in my case the problem was as follows:

- Visit a page normally using Cypress
- Click a button that submits an HTML form (method post)
- The form returns a redirect response that redirects to another URL
  - This URL was in the same protocol, host and port
  - However, this URL was returning a JSON response instead of HTML response. After fixing this to have a redirect into a normal HTML page, the error vanished.

In short: the error message was misleading. It was not a "cross-origin" error. It was a redirect to a non-html page.

**Comment by papb** [comment]
I suspect @lpyfm's situation may be similar, maybe the error message was misleading too and the actual problem was something else.

**Comment by ZachEspositoPCTY** [comment]
For me the problem was apparently an administrator turning on [certain policies](https://docs.cypress.io/guides/references/error-messages#Cypress-detected-policy-settings-on-your-computer-that-may-cause-issues) - running in electron instead of chrome worked
